sklearn/metrics/_plot/tests/test_confusion_matrix_display.py

```diff
@@ -0,0 +1,328 @@
+def data(n_classes):
+    X, y = make_classification(
+        n_samples=100, n_informative=5, n_classes=n_classes, random_state=0
+    )
+    return X, y
+
+
+@pytest.fixture(scope="module")
+def fitted_clf(data):
+    return SVC(kernel="linear", C=0.01).fit(*data)
+
+
+@pytest.fixture(scope="module")
+def y_pred(data, fitted_clf):
+    X, _ = data
+    return fitted_clf.predict(X)
+
+
+@pytest.fixture(scope="module")
+def estimator_api_params(fitted_clf, data):
+    X, y = data
+    return ConfusionMatrixDisplay.from_estimator, fitted_clf, X, y
+
+
+@pytest.fixture(scope="module")
+def predictions_api_params(data, y_pred):
+    _, y = data
+    return ConfusionMatrixDisplay.from_predictions, y, y_pred
+
+
+@pytest.fixture(
+    params=["estimator_api_params", "predictions_api_params"],
+    scope="module",
+)
+def confusion_matrix_display_fxt(request):
+    return request.getfixturevalue(request.param)
+
+
+def test_confusion_matrix_display_error_on_regressor(pyplot, data, y_pred):
+    X, y = data
+    regressor = SVR().fit(X, y)
+
+    err_msg = "ConfusionMatrixDisplay.from_estimator only supports classifiers"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_estimator(regressor, X, y)
+
+    y_pred_regressor = regressor.predict(X)
+
+    err_msg = (
+        "Only 'binary' and 'multiclass' classification problems are supported"
+    )
+    with pytest.raises(ValueError, match=err_msg):
+        # Force `y_true` to be seen as a regression problem
+        ConfusionMatrixDisplay.from_predictions(y + 0.5, y_pred)
+
+    err_msg = "Got `y_pred` of type 'continuous'"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_predictions(y, y_pred_regressor)
+
+    err_msg = "Found input variables with inconsistent numbers of samples"
+    with pytest.raises(ValueError, match=err_msg):
+        ConfusionMatrixDisplay.from_predictions(y, y_pred[::2])
+
+
+def test_confusion_matrix_display_invalid_option(
+    pyplot, confusion_matrix_display_fxt
+):
+    constructor, *params = confusion_matrix_display_fxt
+    extra_params = {"normalize": "invalid"}
+
+    err_msg = r"normalize must be one of \{'true', 'pred', 'all', None\}"
+    with pytest.raises(ValueError, match=err_msg):
+        constructor(*params, **extra_params)
+
+
+@pytest.mark.parametrize("with_labels", [True, False])
+@pytest.mark.parametrize("with_display_labels", [True, False])
+def test_confusion_matrix_display_custom_labels(
+    pyplot,
+    confusion_matrix_display_fxt,
+    data,
+    y_pred,
+    n_classes,
+    with_labels,
+    with_display_labels,
+):
+
+    X, y = data
+    constructor, *params = confusion_matrix_display_fxt
+
+    ax = pyplot.gca()
+    labels = [2, 1, 0, 3, 4] if with_labels else None
+    display_labels = ["b", "d", "a", "e", "f"] if with_display_labels else None
+
+    cm = confusion_matrix(y, y_pred, labels=labels)
+    disp = constructor(
+        *params,
```

Comment:
Regarding the use of fixtures: it's not obvious here at first sight that the parameters passed to `confusion_matrix` (namely y and y_pred) are also somehow contained in `*params`, and thus it's not immediately explicit that `cm` is supposed to be the same as `disp.cm`

Reply:
I agree regarding the usage of fixtures. Somehow they were there and I adapted the code to easily both constructor.
However, I think that here the execution time is not the bottleneck and we could repeat some code which will make the test more self-explained.